R/surv-brier.R

```diff
@@ -0,0 +1,166 @@
+#'     censoring_weights = ipcw,
+#'     eval_time = .time
+#'   )
+#' @export
+brier_survival <- function(data, ...) {
+  UseMethod("brier_survival")
+}
+
+brier_survival <- new_dynamic_survival_metric(
+  brier_survival,
+  direction = "minimize"
+)
+
+#' @rdname brier_survival
+#' @export
+brier_survival.data.frame <- function(data,
+                                      truth,
+                                      estimate,
+                                      censoring_weights,
+                                      eval_time,
+                                      na_rm = TRUE,
+                                      case_weights = NULL,
+                                      ...) {
+  data <- dplyr::group_by(data, {{eval_time}})
+
+  dynamic_survival_metric_summarizer(
+    name = "brier_survival",
+    fn = brier_survival_vec,
+    data = data,
+    truth = !!enquo(truth),
+    estimate = !!enquo(estimate),
+    censoring_weights = !!enquo(censoring_weights),
+    eval_time = !!enquo(eval_time),
+    na_rm = na_rm,
+    case_weights = !!enquo(case_weights)
+  )
+}
+
+#' @export
+#' @rdname brier_survival
+brier_survival_vec <- function(truth,
+                               estimate,
+                               censoring_weights,
+                               eval_time,
+                               na_rm = TRUE,
+                               case_weights = NULL,
+                               ...) {
+  check_dynamic_survival_metric(
+    truth, estimate, censoring_weights, case_weights, eval_time
+  )
+
+  n_distinct_time <- dplyr::n_distinct(eval_time)
+  if (n_distinct_time != 1) {
+    abort(paste0(
+      "`eval_time` should have at most 1 unique value. But ", n_distinct_time,
+      " was detected."
+    ))
+  }
+
+  if (na_rm) {
+    result <- yardstick_remove_missing(
+      truth, estimate, case_weights, censoring_weights, eval_time
+    )
+
+    truth <- result$truth
+    estimate <- result$estimate
+    censoring_weights <- result$censoring_weights
+    eval_time <- result$eval_time
+    case_weights <- result$case_weights
+  } else {
+    any_missing <- yardstick_any_missing(
+      truth, estimate, case_weights, censoring_weights, eval_time
+    )
+    if (any_missing) {
+      return(NA_real_)
+    }
+  }
+
+  brier_survival_impl(truth, estimate, censoring_weights, case_weights, eval_time)
+}
+
+brier_survival_impl <- function(truth,
+                                estimate,
+                                censoring_weights,
+                                case_weights,
+                                eval_time) {
+  surv_time <- truth[, "time"]
+  surv_status <- truth[, "status"]
+
+  if (!is.null(case_weights)) {
+    norm_const <- sum(case_weights)
+  } else {
+    case_weights <- rep(1, length(estimate))
+    norm_const <- sum(!survival::is.na.Surv(truth))
+  }
+
+  category_1 <- surv_time < eval_time & surv_status == 1
+  category_2 <- surv_time >= eval_time
+
+  # (0 - estimate) ^ 2 == estimate ^ 2
```

Comment:
😄

Reply:
It needed to be said, otherwise I would get back in 2 years and have no idea 😆

Reply:
Agree! 😄